src/event-store.ts

```diff
@@ -37,15 +32,36 @@ export interface EventStoreOptions {
   logger?: Logger;
 }
 
-export interface ListenOptions {
-  executeHandlerIfEventExists?: boolean;
-  requestReplay?: boolean;
+/**
+ *  Compose a list of maching functions into an aggregator
+ *
+ */
+export function composeAggregator<T>(matches: AggregateMatches<T>): Aggregator<T> {
```

Comment:
Just a comment. Why reordering the function implementation? I know is inconsistent and I actually favour writing the code this way, always leaving the core entry point of the module in the bottom of the file, I'm used to do it from when I learn C from my cretaceous cs teacher. But avoiding messing up things with function asignations instead of leaving the redorering to hoisting is a good excuse.

I'm just curious if you have any other reason for it.

Reply:
[this reason](https://eslint.org/docs/rules/no-use-before-define)